hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/prune/StaticPartitionPruner.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.source.prune;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Static partition pruner for hoodie table source which partitions list is available in compile phase.
+ * After applied this partition pruner, hoodie source could not read the data from other partitions during runtime.
+ * Note: the data of new partitions created after the job starts would never be read.
+ */
+public class StaticPartitionPruner implements PartitionPruner {
+
+  private static final long serialVersionUID = 1L;
```

Comment:
Can we merge the PartitionPruner into one, it can cache a list of candidate partitions which have higher priority always.

Reply:
In fact, `DynamicPartitionPruner` and `StaticPartitionPruner` could not happen at the same time.
`StaticPartitionPruner` could only work for `HoodieTableSource` because Flink optimizer pushes down satisfied partitions instead of partition pruning filter conditions.
`DynamicPartitionPruner` could only work for `HoodieContinuousPartitionTableSource` because Flink optimizer pushes down partition pruning conditions instead of  satisfied partitions.